### PR TITLE
RAG Benchmarking - LLM as a Judge

### PR DESCRIPTION
**Pull request by sreanikdk**
An example is being added to evaluate the responses from a RAG application (created with RetrievalQA chain) using LLM-as-a-judge approach.

### CODE REVIEW

File: scripts/step03_explore_examples/02_advanced_rag/src/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+log = logging.getLogger(__name__)
+
+class ValidationResult(BaseModel):
+    query: str
+    actual_answer: str
+    rating: float
+    reasoning: str
+
+    class Config:
```
Comment (EvgeniiSkrebtcov) [inline scripts/step03_explore_examples/02_advanced_rag/src/benchmark.py:25]: poetry lock refers to Pydantic V2, where usage of Config is obsolete: 

name = "pydantic"
**version = "2.8.2"**

https://docs.pydantic.dev/2.2/migration/#changes-to-config

File: scripts/step03_explore_examples/02_advanced_rag/src/ingest.py [hunk 2]
```diff
@@ -76,11 +76,43 @@ def ingest_sap_docs():
     db.add_documents(chunks)
     log.success("Added SAP btp docs successfully!")
 
+def load_terraform_docs_from_github():
```
Comment (EvgeniiSkrebtcov) [inline scripts/step03_explore_examples/02_advanced_rag/src/ingest.py:79]: This Function repeats the functionality, please consider moving it to [Utils](https://github.com/SAP-samples/btp-genai-starter-kit/tree/main/scripts/step03_explore_examples/utils/utils): https://github.com/SAP-samples/btp-genai-starter-kit/blob/50e3d13e0a9e9bdf2a7eff4221149986a24cab25/scripts/step03_explore_examples/01_naive_rag/library/ingest.py#L19

File: scripts/step03_explore_examples/02_advanced_rag/src/ingest.py [hunk 3]
```diff
@@ -76,11 +76,43 @@ def ingest_sap_docs():
...
+        branch="main",
+    )
+    text_documents = loader.load()
+    log.info("Getting the Terraform documents from the GitHub repository ...")
+
+    # Split the documents into chunks
+    chunks = split_docs_into_chunks(documents=text_documents)
+    return chunks    
+
+def ingest_terraform_docs():
```
Comment (EvgeniiSkrebtcov) [inline scripts/step03_explore_examples/02_advanced_rag/src/ingest.py:95]: This Function repeats the functionality, please consider moving it to [Utils](https://github.com/SAP-samples/btp-genai-starter-kit/tree/main/scripts/step03_explore_examples/utils/utils): https://github.com/SAP-samples/btp-genai-starter-kit/blob/50e3d13e0a9e9bdf2a7eff4221149986a24cab25/scripts/step03_explore_examples/01_naive_rag/library/ingest.py#L36

File: scripts/step03_explore_examples/02_advanced_rag/src/benchmark.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    llm = ChatOpenAI(
+        proxy_model_name=JUDGE_LLM,
+        proxy_client=proxy_client,
+        temperature=0,
+    )
+
+    try:
+        validation_result = llm.invoke(messages)
+        log.info(validation_result.content)
+        rating_match = re.search(r"Total rating: (\d+(\.\d+)?)", validation_result.content)
```
Comment (EvgeniiSkrebtcov) [inline scripts/step03_explore_examples/02_advanced_rag/src/benchmark.py:95]: There's no handling for cases where the regex doesn't find a match, which could lead to an AttributeError when rating_match.group(1) is accessed.

**Review by EvgeniiSkrebtcov — CHANGES_REQUESTED** [review]

**Comment by sreanikdk** [comment]
Will close this PR and create a new one(incorporating the review comments) with a dedicated set of examples for RAG Benchmarking.

**Pull request closed without merging**